.github/workflows/test.yml

```diff
@@ -0,0 +1,65 @@
+name: Test
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
+  workflow_dispatch:
+    inputs: {}
+
+jobs:
+  tests:
+    strategy:
+      matrix:
+        include:
+        - { python-version: "3.7", os: "ubuntu-latest" }
+        - { python-version: "3.8", os: "ubuntu-latest" }
+        - { python-version: "3.9", os: "ubuntu-latest" }
+      fail-fast: false
+
+    name: Pytest on py${{ matrix.python-version }} (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    env:
+      PYTEST_MARKERS: not concurrent
+
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v3.0.2
+
+    - name: Install Poetry
+      env:
+        PIP_CONSTRAINT: .github/workflows/constraints.txt
+      run: |
+        pipx install poetry
+        poetry --version
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3.1.2
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+        cache: 'poetry'
+
+    - name: Upgrade pip
+      env:
+        PIP_CONSTRAINT: .github/workflows/constraints.txt
+      run: |
+        pip install pip
+        pip --version
+
+    - name: Install Dependencies
+      run: |
+        poetry env use "${{ matrix.python-version }}"
+        poetry install
+
+    - name: Run pytest
+      run: |
+        poetry run pytest -v --cov-report= --cov meltano -m "$PYTEST_MARKERS"
```

Comment:
Also, it might be worth computing coverage in all our supported Python versions and then combine them in a single report, like we do in the SDK: 

https://github.com/meltano/sdk/blob/db90479f01d0426c380dd7d1b0234d4fbc9ec8e4/tox.ini#L47

Reply:
@pandemicsyn re:

> do you wanna disable the pytest on the gitlab side to cut down our build minute usage there or leave that enabled for a bit?

I think I'll hold off on removing those for now - just because I want to discuss the docker-image-dependent pytests and if we still need those. The `pytest_fast` could be dropped but I still like having that one as quick smoke check that doesn't take 20+ minutes to give a signal. I think we'll end up dropping them but want to make sure it's a thought-through decision.